### PR TITLE
[msg] Retry state for failed messages

### PR DESCRIPTION
**Pull request by sarahscott**
If a message mutation fails, the message text will show up in the Composer again so the user can try again

Fixes: https://github.com/artsy/emission/issues/625

### CODE REVIEW

File: src/lib/Containers/Conversation.tsx [hunk 1]
```diff
@@ -127,9 +128,10 @@ export class Conversation extends React.Component<Props, State> {
 
     return (
       <Composer
-        disabled={this.state.sendingMessage}
+        disabled={this.state.sendingMessage || !this.state.isConnected}
+        value={this.state.submissionError}
```
Comment (alloy) [inline src/lib/Containers/Conversation.tsx]: I’m not a big fan of this name, I wonder if it could be something like `failedMessageText` ?

**Comment by sarahscott** [comment]
oh good catch! i totally agree and will update.

**Comment by mennenia** [comment]
@sarahscott looks awesome! Wanna merge the conflicts with master? Then we can 🚢  🎉 !! :D

File: src/lib/Components/Inbox/Conversations/Composer.tsx [hunk 2]
```diff
@@ -74,6 +80,8 @@ export default class Composer extends React.Component<Props, State> {
       paddingRight: 10,
     }
 
+    const disableSendButton = !(this.state.text && this.state.text.length) || this.props.disabled
```
Comment (alloy) [inline src/lib/Components/Inbox/Conversations/Composer.tsx]: FYI I’d generally revert this to _first_ check the simple property and only if that doesn’t short-circuit the conditional do some computation.

**Pull request merged**